### PR TITLE
@/roboshild CMS Improvements

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

This PR improves our use of CMS, specifically:

  1. Use CMS to store and retrieve analytics code rather than ENV var. This should remove Tech as blocker when it comes to analytics i.e. PM should be able to update and manage analytics moving forward
  2. Use Analytics to track important events such as download for robots.txt file, etc.
  3. Incorporate ability to fetch draft pages/posts. This is only restricted to the Next.js draft mode i.e. PREVIEW in the CMS (Not sure if the `useLivePreview` plugin uses draft mode underneath or not)
  4. Streamline merging of SEO information (Post -> Page -> Site) to `next-seo`.

All this is first implemented in RoboShield for review with plans to roll changes to other CMS-driven apps once the PR has been reviewed and accepted by @CodeForAfrica/tech .

-----

**Original Fix**

While `useLivePreview` update `data` on page navigation, the page (specifically, the `main` part of page layout) doesn't get updated/re-rendered to reflect new data. This PR introduces `slug` prop that will tell React that a page has changed and `main` needs to be re-rendered.

By tracking this in the layout component (i.e. `Page`), the rest of the app doesn't need be updated (including `useLivePreview` hook itself).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Tracking of Download via Analytics

Screenshot 2024-08-16 at 16 36 27

### CMS with Analytics Screen

Screenshot 2024-08-16 at 15 11 25

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by kilemensi** [comment]
@koechkevin, ☝🏽  is my understanding of the issues. Happy for us to discuss and compare with #840

**Comment by kilemensi** [comment]
@koechkevin / @m453h ?

**Comment by koechkevin** [comment]
> @koechkevin / @m453h ?

Reviewing

**Review by koechkevin — APPROVED** [review]
This works @kilemensi specifically for  Roboshield.

I say this because we do not do much page processing on the serverside on Roboshield unlike the other apps.

**Comment by kilemensi** [comment]
>I say this because we do not do much page processing on the serverside on Roboshield unlike the other apps.
>

As long as we update the slug, I think it should work on any app @koechkevin. Is there an app that you're specifically worried about that I should try it on?

**Comment by koechkevin** [comment]
> > I say this because we do not do much page processing on the serverside on Roboshield unlike the other apps.
> 
> As long as we update the slug, I think it should work on any app @koechkevin. Is there an app that you're specifically worried about that I should try it on?

I mean the whole `useLivePreview` functionality @kilemensi. 
PayloadCMS live preview works by listening to changes in the current document and emits an event with the new unsaved document to frontend component. In apps such as charterafrica, we process some pages and blocks on the serverside before returning processed props to the component which under the hood executes `findPage` which doesn't get the latest state of the document. This will work on the published page but won't work on the preview in admin panel.

**Review by m453h — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -387,9 +387,12 @@ ARG NEXT_PUBLIC_APP_LOGO_URL \
   PAYLOAD_PUBLIC_APP_URL
 
 # TODO(koech): Standadise naming of GA MEASUREMENT ID. Our options:
-#              - GA_MEASUREMENT_ID (charterafrica, civicsignalblog, codeforafrica)
-#              - GOOGLE_ANALYTICS_ID (pesayetu, roboshield, vpnmanager)
+#              - GA_MEASUREMENT_ID (charterafrica)
+#              - GOOGLE_ANALYTICS_ID (pesayetu, vpnmanager)
 #              This is only needed at runtime
+# TODO(koech): Completely remove the use of ENV var for Googel analytics
```
Comment (kilemensi) [inline Dockerfile:393]:
```suggestion
# TODO(koech): Completely remove the use of ENV vars for Google Analytics
```

**Review by kelvinkipruto — APPROVED** [review]

**Review by kelvinkipruto — APPROVED** [review]
LGTM

**Pull request merged**